src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -82,23 +82,33 @@ export default function IndexPatternFactory(Private, Notifier, config, kbnIndex,
     // give index pattern all of the values in _source
     _.assign(indexPattern, response._source);
 
-    indexFields(indexPattern);
+    const promise = indexFields(indexPattern);
 
     // any time index pattern in ES is updated, update index pattern object
     docSources
     .get(indexPattern)
     .onUpdate()
     .then(response => updateFromElasticSearch(indexPattern, response), notify.fatal);
+
+    return promise;
+  }
+
+  function containsFieldStats(fields) {
+    return _.any(fields, (field) => {
+      return _.has(field, 'aggregatable') && _.has(field, 'searchable');
+    });
   }
 
   function indexFields(indexPattern) {
     if (!indexPattern.id) {
       return;
     }
-    if (!indexPattern.fields) {
-      return indexPattern.refreshFields();
+
+    let promise = Promise.resolve();
```

Comment:
Lets move this to the top of the function so that the return on 104 can also return a promise.